Tagbuilds

andrew-m-leonard opened the pull request:
Provides tag driven builds for all jdk versions, main & evaluation
Features:
- Triggers when a new "unpublished" upstream build source tag appears eg.jdk-11.0.23+2
- OverridePublishName set to "\<tag\>-ea"
- Published binaries release name set to "\<tag\>-ea-beta"
- Artifact filenames contains publish name pruned as per standard release naming,eg.11.0.23_2, or 8u412b03
- Main & Evaluation pipelines triggered
- Option to "Force" Main or Evaluation even if tag already exists, for example to re-build a missing platform
- "Force" override of targetConfigurations, so that force of a single failed platform can be submitted
- Old scheduled builds disabled
- Daily Slack status updated to support tag driven builds for all versions

Test job: https://ci.adoptium.net/job/build-scripts/job/utils/job/betaTrigger_8ea/18/parameters/

Fixes: https://github.com/adoptium/ci-jenkins-pipelines/issues/902

andrew-m-leonard commented:
run tests

andrew-m-leonard commented:
run tests

andrew-m-leonard commented:
run tests

karianna commented:
pipeline-build-check failed in part because the jdk8 pipeline was on a host that ran out of space.

andrew-m-leonard commented:
> pipeline-build-check failed in part because the jdk8 pipeline was on a host that ran out of space.

yep, I think @steelhead31 's hopefully fixed it, i'll re-trigger

andrew-m-leonard commented:
run tests

karianna reviewed (changes requested):
Not so sure on the commented out code, would prefer to see it deleted I think (Git will let us get it back)

andrew-m-leonard commented:
> Not so sure on the commented out code, would prefer to see it deleted I think (Git will let us get it back)

@karianna yes, agree, better to delete the lines

andrew-m-leonard commented:
run tests

andrew-m-leonard commented:
run tests

sxa reviewed (commented):
Initial comment before I properly analyse the code - can we add some end-user information on this new process into https://github.com/adoptium/ci-jenkins-pipelines/blob/master/README.md where we describe the current nightly triggering.

sxa commented:
> > Not so sure on the commented out code, would prefer to see it deleted I think (Git will let us get it back)
> 
> @karianna yes, agree, better to delete the lines

In this case I'd actually prefer to have them present and commented out so it's obvious where the time-based scheduling can be re-enabled if desired. We will presumably still need this functionality for new releases while they are maintained by Oracle and not getting regular tags.

I guess as a middle ground we could just leave it in the JDK22+ ones if you don't want to retain it for all of them.

sxa reviewed (commented):
The logic SGTM subject to my limited groovy expertise ... Most of the things in here are queries and doc suggestions that it would be nice to have answers to prior to giving a formal approval but it looks like you've done some testing on it so it should be good. We'll see if there are any other edge cases we've missed once it goes live, but this should be a lot nicer than my original prototype code that we've been relying on for a few months!

andrew-m-leonard commented:
> Initial comment before I properly analyse the code - can we add some end-user information on this new process into https://github.com/adoptium/ci-jenkins-pipelines/blob/master/README.md where we describe the current nightly triggering.

thanks for pointing out, yes updated

> Initial comment before I properly analyse the code - can we add some end-user information on this new process into https://github.com/adoptium/ci-jenkins-pipelines/blob/master/README.md where we describe the current nightly triggering.

thanks for pointing out, updated and added description of pipeline groovy and parameters

andrew-m-leonard commented:
> > > Not so sure on the commented out code, would prefer to see it deleted I think (Git will let us get it back)
> > 
> > 
> > @karianna yes, agree, better to delete the lines
> 
> In this case I'd actually prefer to have them present and commented out so it's obvious where the time-based scheduling can be re-enabled if desired. We will presumably still need this functionality for new releases while they are maintained by Oracle and not getting regular tags.
> 
> I guess as a middle ground we could just leave it in the JDK22+ ones if you don't want to retain it for all of them.

Yes, i've added back in as a comment for jdk22.groovy, which will be used as a basis for jdk22u.groovy which will required the schedule

sxa commented:
Raised [question in slack](https://adoptium.slack.com/archives/C09NW3L2J/p1707305735196349?thread_ts=1707235495.875719&cid=C09NW3L2J) about whether there might be a better name than `main` for the non-evaluation pipelines (especially since `main` is now used by github as the default branch in new repositories) but that shouldn't be considered a blocker from my pespective

andrew-m-leonard commented on pipelines/build/common/build_base_file.groovy:
```diff
@@ -783,33 +783,14 @@ class Builder implements Serializable {
             tag = publishName
         }
 
-        // Definition of filenames when building an EA tag. This is passed
```
you're welcome :-)

andrew-m-leonard commented on pipelines/build/common/trigger_beta_build.groovy:
```diff
@@ -0,0 +1,143 @@
+
+    if (triggerMainBuild || triggerEvaluationBuild) {
+        // Set version suffix, jdk8 has different mechanism to jdk11+
+        def additionalConfigureArgs =  (version > 8) ? "--with-version-opt=ea" : "--with-milestone=beta"
```
yeah I specified it for jdk8 just so it's clear, but I think you're right

andrew-m-leonard commented on pipelines/build/common/trigger_beta_build.groovy:
```diff
@@ -0,0 +1,143 @@
+    }
+    echo "Latest Adoptium tag from ${mirrorRepo} = ${latestAdoptTag}"
+
+    def buildTag=latestAdoptTag.replaceAll("_adopt","-ea-beta")
```
yeah, i'm changing that to be clearer. Build tag is the binaries repo release tag. It is in fact the publishJobTag which then has "-beta" appended by the publish job: https://github.com/adoptium/github-release-scripts/blob/2e8dea702fda98c23a52eb436d1927261d29b0d1/sbin/Release.sh#L165

karianna commented on README.md:
```diff
@@ -151,6 +151,12 @@ disableJob = true
 
 #### triggerSchedule_nightly / triggerSchedule_weekly / triggerSchedule_evaluation / triggerSchedule_weekly_evaluation
 
+All JDK versions now support "beta" EA triggered builds from the publication of upstream build tags. Eclipse Adoptium no
```
had a typo so I didn't commit that.
